File: src-tauri/src/models/folder.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Folder {
    pub id: String,
    pub path: String,
    pub name: String,
    pub media_count: usize,
    pub added_at: String,
    pub last_scanned: Option<String>,
}

File: src-tauri/resources/server/src/types/folder.ts


// Folder Interface
export interface Folder {
  id: string;
  path: string;
  name: string;
  mediaCount: number;
  addedAt: string;
  lastScanned?: string;
}
export interface FolderPreview {
  path: string;
  name: string;
  fileCount: number;
  estimatedMovies: number;
  estimatedSeries: number;
}